internal/webhook/apps/v1alpha1/nimservice_webhook_validation_helper.go

```diff
@@ -0,0 +1,251 @@
+	}
+	return errList
+}
+
+// If Spec.Expose.Ingress.Enabled is true, Spec.Expose.Ingress.Spec must be non-nil.
+func validateExposeConfiguration(expose *appsv1alpha1.Expose, fldPath *field.Path) field.ErrorList {
+	errList := field.ErrorList{}
+	if expose.Ingress.Enabled != nil && *expose.Ingress.Enabled {
+		if reflect.DeepEqual(expose.Ingress.Spec, networkingv1.IngressSpec{}) {
+			errList = append(errList, field.Required(fldPath.Child("spec"), fmt.Sprintf("must be defined if %s is true", fldPath.Child("enabled"))))
+		}
+	}
+	return errList
+}
+
+// If Spec.Metrics.Enabled is true, Spec.Metrics.ServiceMonitor must not be empty.
+func validateMetricsConfiguration(metrics *appsv1alpha1.Metrics, fldPath *field.Path) field.ErrorList {
+	errList := field.ErrorList{}
+	if metrics.Enabled != nil && *metrics.Enabled {
+		if reflect.DeepEqual(metrics.ServiceMonitor, appsv1alpha1.ServiceMonitor{}) {
+			errList = append(errList, field.Required(fldPath.Child("serviceMonitor"), fmt.Sprintf("must be defined if %s is true", fldPath.Child("enabled"))))
+		}
+	}
+	return errList
+}
+
+// If Spec.Scale.Enabled is true, Spec.Scale.HPA must be non-empty HorizontalPodAutoScaler.
+func validateScaleConfiguration(scale *appsv1alpha1.Autoscaling, fldPath *field.Path) field.ErrorList {
+	errList := field.ErrorList{}
+	if scale.Enabled != nil && *scale.Enabled {
+		if reflect.DeepEqual(scale.HPA, appsv1alpha1.HorizontalPodAutoscalerSpec{}) {
+			errList = append(errList, field.Required(fldPath.Child("hpa"), fmt.Sprintf("must be defined if %s is true", fldPath.Child("enabled"))))
+		}
+	}
+	return errList
+}
+
+// Spec.DRAResources must be empty unless K8s version is v1.33.1+.
+func validateDRAResourcesVersionCompatibility(kubeVersion string, fldPath *field.Path) field.ErrorList {
+
+	errList := field.ErrorList{}
+
+	if !utils.IsVersionGreaterThanOrEqual(kubeVersion, utils.MinSupportedClusterVersionForDRA) {
+		errList = append(errList, field.Forbidden(fldPath, fmt.Sprintf("is not supported by NIM-Operator on this cluster, please upgrade to k8s version '%s' or higher", utils.MinSupportedClusterVersionForDRA)))
+	}
+
+	return errList
+}
+
+// Spec.Resources.Claims must be empty.
+func validateResourcesConfiguration(resources *corev1.ResourceRequirements, fldPath *field.Path) field.ErrorList {
+	errList := field.ErrorList{}
+	if resources != nil {
+		if resources.Claims != nil || len(resources.Claims) != 0 {
+			errList = append(errList, field.Forbidden(fldPath.Child("claims"), "must be empty"))
+		}
+	}
+	return errList
+}
+
+// validateNIMServiceSpec aggregates all structural validation checks for a NIMService
+// object. It is intended to be invoked by both ValidateCreate and ValidateUpdate to
+// ensure the resource is well-formed before any other validation (e.g. immutability)
+// is performed.
+func validateNIMServiceSpec(kubeVersion string, spec *appsv1alpha1.NIMServiceSpec, fldPath *field.Path) field.ErrorList {
```

Comment:
nit: reorder arg list
```suggestion
func validateNIMServiceSpec(spec *appsv1alpha1.NIMServiceSpec, fldPath *field.Path, k8sVersion string) field.ErrorList {
```